src/pages/index.js

```diff
@@ -54,26 +124,77 @@ addButton.addEventListener("click", () => {
   addNewCardFormValidator.disableSubmitButton();
   addNewCardModal.open();
 });
+updateAvatarIcon.addEventListener("click", () => {
+  updateAvatarFormValidator.disableSubmitButton();
+  updateAvatarModal.open();
+});
 
 export const userInfo = new UserInfo({
   userNameSelector: ".profile__heading",
   userOccupationSelector: ".profile__sub-heading",
+  profileAvatarSelector: ".profile__avatar",
 });
 
 export function updateProfile(profileObject) {
   const newName = profileObject["input-name"];
   const newOccupation = profileObject["input-description"];
+  console.log(profileObject);
+  profileSubmitButton.textContent = "Saving...";
+
+  api
+    .updateProfile({
+      name: newName,
+      occupation: newOccupation,
+    })
+    .finally(() => {
+      profileSubmitButton.textContent = "Save";
+    });
   userInfo.setUserInfo(newName, newOccupation);
   editProfileFormValidator.disableSubmitButton();
   editProfileModal.close();
 }
 
+export function updateProfilePicture(profileObject) {
+  const newPhotoLink = profileObject["input-url"];
+  console.log(profileObject);
+  changeAvatarSubmitButton.textContent = "Saving...";
+  api.updateAvatar(newPhotoLink).then((response) => {
+    userInfo.setUserAvatar(response.avatar);
+    changeAvatarSubmitButton.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.